Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs

```diff
@@ -673,6 +673,23 @@ public async Task<SandboxedProcessPipExecutionResult> RunAsync(CancellationToken
                     string arguments = m_pip.Arguments.ToString(m_pipDataRenderer);
                     m_timeout = GetEffectiveTimeout(m_pip.Timeout, m_sandboxConfig.DefaultTimeout, m_sandboxConfig.TimeoutMultiplier);
 
+                    // If ChangeAffectedInputListWrittenFilePath is set, write the change affected inputs of the pip to the file.

```

Comment:
>// If ChangeAffectedInputListWrittenFilePath is set, write the change affected inputs of the pip to the file. [](start = 20, length = 109)

Refactor this part into a method. See `PrepareResponseFile` for a reference. You just have to create a similar method to that.